### PR TITLE
Added ADR for shell api

### PR DESCRIPTION
**Pull request by ZKRobi**

### CODE REVIEW

File: architecture/adr-016-shell-api.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+| Type          | ID            | Status        | Title                   |
+| ------------- | ------------- | ------------- | ----------------------- |
+| ADR           | adr-016       | Approved      | Shell API               |
+
+# Architecture Decision Record: Shell API
+
+## Context
+
+We must ensure apps written for Compose UI can run in any shell implementation and provide a consistent developer experience across shells. In order to achieve this, a fixed set of features for shell interaction need to be defined and made available for developers.
```
Comment (ztanczos) [inline architecture/adr-016-shell-api.md:11]: Nitpick: `ComposeUI`

File: architecture/adr-016-shell-api.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+We wish to ensure apps written for ComposeUI can run in any shell implementation and provide a consistent developer experience across shells. In order to achieve this, a fixed set of features for shell interaction need to be defined and made available for developers.
+
+Another consideration we must make is how easy is it to bring existing applications into the ComposeUI environment. "Basic" web apps are expected to run out of the box, but apps targeted at running inside containers might leverage specific APIs. We need to consider API compatibility with other desktop containers as well as DesktopJS, an abstraction layer over multiple web container APIs we are maintaining.
+
+### A list of features we want to provide with high priority:
+#### Window management
+Opening and closing of child windows with the opportunity to customize their properties like size, position, available controls, "topmostness" etc.
+Child windows will need to fit the style of the Shell.
+Initially these windows will be only part of the app state, but in the future interaction with layout saving will need to be considered.
```
Comment (ztanczos) [inline architecture/adr-016-shell-api.md:19]: Can you please explain what do you mean by 'will be only part of the app state'?

File: architecture/adr-016-shell-api.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+#### IPC
+We already provide IPC functionality via the MessageRouter API and via FDC3. These fit our current IPC needs.
+
+#### Logging
+Centralized logging to a remote location is a highly desired feature for large organizations. 
+ 
+### Future features we consider:
+#### Management of chrome elements
+Running apps might want to add dynamic functionality to the chrome (e.g. custom contextual ribbon buttons or shortcuts to certain functionality). 
+#### Opening apps (Currently a priority via fdc3)
```
Comment (ztanczos) [inline architecture/adr-016-shell-api.md:39]: You use 'app' and 'module' interchangeably here. Which one do you mean?

File: architecture/adr-016-shell-api.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+We wish to ensure apps written for ComposeUI can run in any shell implementation and provide a consistent developer experience across shells. In order to achieve this, a fixed set of features for shell interaction need to be defined and made available for developers.
+
+Another consideration we must make is how easy is it to bring existing applications into the ComposeUI environment. "Basic" web apps are expected to run out of the box, but apps targeted at running inside containers might leverage specific APIs. We need to consider API compatibility with other desktop containers as well as DesktopJS, an abstraction layer over multiple web container APIs we are maintaining.
+
+### A list of features we want to provide with high priority:
+#### Window management
+Opening and closing of child windows with the opportunity to customize their properties like size, position, available controls, "topmostness" etc.
+Child windows will need to fit the style of the Shell.
+Initially windows opened by modules are considered to be only part of the app state and thus will not be saved as part of the layout. In the future interaction with layout saving will need to be considered.
```
Comment (ztanczos) [inline architecture/adr-016-shell-api.md:19]: > Initially windows opened by modules are considered to be only part of the app state and thus will not be saved as part of the layout.

In [ADR-013](https://github.com/morganstanley/ComposeUI/blob/main/architecture/adr-013-script-preloading-security.md) (which is in Draft) the decision was that
> ... popup windows and iframes will not inherit the parent window's module identity or injected scripts. When a module wants to pop up another window, it has to do so using container APIs (eg. FDC3 raiseIntent or open, or the Module Loader start API).

This to me means that windows opened by modules are new apps, cannot be part of the app's state which opened them. Am I missing something?

**Review by ztanczos — APPROVED** [review]

**Review by lilla28 — APPROVED** [review]

**Pull request merged**